scipy/stats/stats.py

```diff
@@ -4259,6 +4265,529 @@ def weightedtau(x, y, rank=True, weigher=None, additive=True):
+
+    if nx < 5 or ny < 5:
+        raise ValueError("MGC requires at least 5 samples to give reasonable "
+                         "results.")
+
+    # convert x and y to float
+    x = np.asarray(x).astype(np.float64)
+    y = np.asarray(y).astype(np.float64)
+
+    # check if compute_distance_matrix if a callable()
+    if not callable(compute_distance) and compute_distance is not None:
+        raise ValueError("Compute_distance must be a function.")
+
+    # check if number of reps exists, integer, or > 0 (if under 1000 raises
+    # warning)
+    if not isinstance(reps, int) or reps < 0:
+        raise ValueError("Number of reps must be an integer greater than 0.")
+    elif reps < 1000:
+        msg = ("The number of replications is low (under 1000), and p-value "
+               "calculations may be unreliable. Use the p-value result, with "
+               "caution!")
+        warnings.warn(msg, RuntimeWarning)
+
+    if is_twosamp:
+        x, y = _two_sample_transform(x, y)
+
+    # calculate MGC stat
+    stat, stat_dict = _mgc_stat(x, y, compute_distance)
+    stat_mgc_map = stat_dict["stat_mgc_map"]
+    opt_scale = stat_dict["opt_scale"]
+
+    # calculate permutation MGC p-value
+    pvalue, null_dist = _perm_test(x, y, stat, compute_distance, reps=reps,
+                                   workers=workers)
+
+    # save all stats (other than stat/p-value) in dictionary
+    mgc_dict = {"mgc_map": stat_mgc_map,
+                "opt_scale": opt_scale,
+                "null_dist": null_dist}
+
+    return MGCResult(stat, pvalue, mgc_dict)
+
+
+def _mgc_stat(x, y, compute_distance):
+    r"""
+    Helper function that calculates the MGC stat. See above for use.
+
+    Parameters
+    ----------
+    x, y : ndarray
+        `x` and `y` have shapes `(n, p)` and `(n, q)` or `(n, n)` and `(n, n)`
+        if distance matrices.
+    compute_distance : callable
+        A function that computes the distance or similarity among the samples
+        within each data matrix. Set to `None` if `x` and `y` are already
+        distance.
+
+    Returns
+    -------
+    stat : float
+        The sample MGC test statistic within `[-1, 1]`.
+    stat_dict : dict
+        Contains additional useful additional returns containing the following
+        keys:
+            - stat_mgc_map : ndarray
+                MGC-map of the statistics.
+            - opt_scale : (float, float)
+                The estimated optimal scale as a `(x, y)` pair.
+    """
+    # set distx and disty to x and y when compute_distance = None
+    distx = x
+    disty = y
+
+    if compute_distance is not None:
+        # compute distance matrices for x and y
+        distx = compute_distance(x)
+        disty = compute_distance(y)
+
+    # calculate MGC map and optimal scale
+    stat_mgc_map = _local_correlations(distx, disty, global_corr='mgc')[0]
+
+    n, m = stat_mgc_map.shape
+    if m == 1 or n == 1:
+        stat = stat_mgc_map[m - 1][n - 1]
```

Comment:
this deserves a comment. why is `stat` the last element of the array if `stat_mgc_map` is a 1-D array?

Reply:
https://github.com/scipy/scipy/pull/10524/commits/fabfbb13356c806099a19d3ce7ed5448ba2219d0